From b44d1d4e4ba77f1d9cf7de73a3bd210d02b5ff36 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 24 Mar 2021 17:26:53 -0400
Subject: [PATCH] Use uintmax_t for handling rlim_t

rlimit types are not representable with long or long long
formats, therefore use uintmax_t to typecast rlim_t types
and use %j to print it

Refit patch for open-vm-tools-11.2.5.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Randy MacLeod <Randy.MacLeod@windriver.com>
Signed-off-by: Trevor Gamblin <trevor.gamblin@windriver.com>
---
 open-vm-tools/lib/hgfsServer/hgfsServerLinux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/open-vm-tools/lib/hgfsServer/hgfsServerLinux.c b/open-vm-tools/lib/hgfsServer/hgfsServerLinux.c
index 87d419a7b..f5cc80b76 100644
--- a/open-vm-tools/lib/hgfsServer/hgfsServerLinux.c
+++ b/open-vm-tools/lib/hgfsServer/hgfsServerLinux.c
@@ -5301,8 +5301,8 @@ HgfsWriteCheckIORange(off_t offset,         // IN:
       goto exit;
    }
 
-   LOG(6, "%s: File Size limits: 0x%"FMT64"x 0x%"FMT64"x\n",
-       __FUNCTION__, fileSize.rlim_cur, fileSize.rlim_max);
+   //LOG(6, ("%s: File Size limits: 0x%jx 0x%jx\n",
+   //    __FUNCTION__, (uintmax_t)fileSize.rlim_cur, (uintmax_t)fileSize.rlim_max));
 
    /*
     * Check the offset is within the file size range.
-- 
2.25.1

